### ISSUE TITLE
MatTooltip: element description duplicates it's accessible name

### ISSUE THREAD

**Issue by svetsem** [angular/components#29503]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Currently, AriaDescriber doesn't set descriptions if they are exactly the same as the aria-label of the element because screen readers will end up reading the same text twice in a row. However, it doesn't cover the cases when the accessibleName is set from the text content or aria-labelledby, so the duplication still happens.

### Reproduction

StackBlitz link: https://stackblitz.com/edit/stackblitz-starters-nhj9lq?file=src%2Fmain.ts
Steps to reproduce:

Use a screen reader to hear the icon button's descriptions, which are the same as the accessible name (for the first two examples; third works as expected).

### Expected Behavior

A screen reader doesn't read the element's description if it's exactly the same as its accessible name.

### Actual Behavior

A screen reader reads the element's description if it's exactly the same as its accessible name, which leads the user to hear the same text twice. Only using aria-label to determine the accessible name allows avoiding this.

### Environment

CDK/Material: 18.1.2
Browser(s): Safari, Chrome
Operating System: macOS 14.2.1
Screen reader: VoiceOver

**Comment by crisbeto** [comment]
Handling this on our end would require monitoring the DOM for text changes which will degrade the performance. I'm closing the issue since it can be easily worked around by the user.
